services/src/main/java/org/keycloak/services/resources/LoginActionsService.java

```diff
@@ -380,31 +379,43 @@ public Response resetCredentialsGET(@QueryParam(AUTH_SESSION_ID) String authSess
             if (!realm.isResetPasswordAllowed()) {
                 event.event(EventType.RESET_PASSWORD);
                 event.error(Errors.NOT_ALLOWED);
-                return ErrorPage.error(session, authSession, Response.Status.BAD_REQUEST, Messages.RESET_CREDENTIAL_NOT_ALLOWED);
+                return ErrorPage.error(session, null, Response.Status.BAD_REQUEST, Messages.RESET_CREDENTIAL_NOT_ALLOWED);
 
             }
-            authSession = createAuthenticationSessionForClient();
+            authSession = createAuthenticationSessionForClient(clientId);
             return processResetCredentials(false, null, authSession, null);
         }
 
         event.event(EventType.RESET_PASSWORD);
         return resetCredentials(authSessionId, code, execution, clientId, tabId);
     }
 
-    AuthenticationSessionModel createAuthenticationSessionForClient()
-      throws UriBuilderException, IllegalArgumentException {
+    AuthenticationSessionModel createAuthenticationSessionForClient(String clientID)
+            throws UriBuilderException, IllegalArgumentException {
         AuthenticationSessionModel authSession;
 
-        // set up the account service as the endpoint to call.
-        ClientModel client = SystemClientUtil.getSystemClient(realm);
+        ClientModel client = session.realms().getClientByClientId(clientID, realm);
+        String redirectUri = null;
+
+        if (client == null) {
+            client = SystemClientUtil.getSystemClient(realm);
+        } else {
+            redirectUri = client.getRedirectUris()
+                    .stream()
+                    .findFirst()
+                    .orElse(null);
+        }
+
+        if (redirectUri == null) {
+            redirectUri = Urls.accountBase(session.getContext().getUri().getBaseUri()).path("/").build(realm.getName()).toString();
```

Comment:
I believe the lines 409-411 should be removed and line 410 should be moved after line 401 - so that redirectUri is set to the redirectUri of account client just in case that it corresponds to account client? Probably something like this:
```
       if (client == null) {
            client = SystemClientUtil.getSystemClient(realm);
            redirectUri = Urls.accountBase(session.getContext().getUri().getBaseUri()).path("/").build(realm.getName()).toString();
        }
````

WDYT?

Reply:
@mposolda I've been thinking about it a little. I know, each client should has at least one redirect URI, but there might be some case, when a client has not. When a client is created, redirect URI doesn't have to be provided, so far. Of course, user is informed about this in some way, but this approach, I think, is resistant to this. So, if the client exists and doesn't have any redirect URI, it's redirected in a same way like client, which is null.

But, if you think, it's unnecessary, I'll remove this. :)

Reply:
+1 to remove this and move after line 401. The thing is, that with the current code, it can happen that "redirectUri" is set to the redirectUri of the account client, even if client_id is different than account. This can have various side-effects. If the null redirectUri is the issue, then it might be better to throw the exception in case of null redirectUri

Reply:
Ok, I agree, but I've just thought it would be better to redirect to account console, than show to user some message "We're sorry..." :)
Thanks @mposolda !